preprocess/process_dicom.py

```diff
@@ -0,0 +1,294 @@
+    inputFileName = ''
+    if len(args) != 0:
+        # try to find a type suitable for the primary input provided
+        inputFileName = args[0]
+    elif set(primaryInputMethods).intersection(kwargs.keys()):
+        for method in primaryInputMethods:
+            if method in kwargs:
+                inputFileName = kwargs[method]
+                break
+    if not inputFileName:
+        raise RuntimeError("No FileName specified.")
+    import itk
+    if "ImageIO" in kwargs:
+        imageIO = kwargs["ImageIO"]
+    else:
+        imageIO = itk.ImageIOFactory.CreateImageIO(
+            inputFileName[0], itk.ImageIOFactory.ReadMode)
+    if not imageIO:
+        raise RuntimeError(
+            "No ImageIO is registered to handle the given file.")
+    componentTypeDic = {"float": itk.F, "double": itk.D,
+                        "unsigned_char": itk.UC, "unsigned_short": itk.US,
+                        "unsigned_int": itk.UI, "unsigned_long": itk.UL,
+                        "unsigned_long_long": itk.ULL, "char": itk.SC,
+                        "short": itk.SS, "int": itk.SI, "long": itk.SL,
+                        "long_long": itk.SLL
+                        }
+    # Read the metadata from the image file.
+    imageIO.SetFileName(inputFileName[0])
+    imageIO.ReadImageInformation()
+    dimension = imageIO.GetNumberOfDimensions()
+    componentAsString = imageIO.GetComponentTypeAsString(
+        imageIO.GetComponentType())
+    component = componentTypeDic[componentAsString]
+    pixel = imageIO.GetPixelTypeAsString(imageIO.GetPixelType())
+    numberOfComponents = imageIO.GetNumberOfComponents()
+    PixelType = itkTemplate.itkTemplate._pixelTypeFromIO(
+        pixel, component, numberOfComponents)
+    ImageType = itk.Image[PixelType, dimension]
+    ReaderType = itk.ImageSeriesReader[ImageType]
+    return ReaderType.New(*args, **kwargs)
+
+
+def get_filenames(in_dir):
+    nameGenerator = itk.GDCMSeriesFileNames.New()
+    # In case multiple series. Probably not necessary
+    nameGenerator.SetUseSeriesDetails(True)
+    # Limit series by date
+    nameGenerator.AddSeriesRestriction("0008|0021")
+
+    nameGenerator.SetDirectory(in_dir)
+    # TODO: See if it is better to keep only the first series or throw
+    # an error message if multiple series are found.
+    seriesIdentifier = nameGenerator.GetSeriesUIDs()
+    fileNames = nameGenerator.GetFileNames(seriesIdentifier[0])
+    return fileNames
+
+
+def rescale_dicom_image_intensity(image, meta_dict):
+    try:
+        # '0028|1050' is the DICOM window Center for display (string). It is
+        # used to apply an intensity windowing filter.
+        center = int(meta_dict['0028|1050'])
+        # '0028|1050' is the DICOM window Width for display (string). It is
+        # used to apply an intensity windowing filter.
+        width = int(meta_dict['0028|1051'])
+    except KeyError:
+        raise Exception("DICOM window `center` or `width` tag not found.")
```

Comment:
In case the window / level is not in the DICOM tags (`center` and `width` will not be defined below), a `RescaleIntensityImageFilter` could be used. This will ensure a mapping to 0 to 255 from what may be a broader range.

Reply:
The new commit should address your concerns.